cmd/charm/charmcmd/push.go

```diff
@@ -110,10 +110,7 @@ func (c *pushCommand) Run(ctxt *cmd.Context) error {
 	defer client.jar.Save()
 
 	// Retrieve the source directory where the charm or bundle lives.
-	srcDir, err := filepath.Abs(c.srcDir)
-	if err != nil {
-		return errgo.Notef(err, "cannot access charm or bundle")
-	}
+	srcDir := ctxt.AbsPath(c.srcDir)
```

Comment:
Changed main to use cmd.DefaultContext so we'll always get an absolute path in the context.